projects/packages/search/compatibility/unsupported-browsers.php

```diff
@@ -0,0 +1,25 @@
+<?php // phpcs:ignore WordPress.Files.FileName.NotHyphenatedLowercase
+/**
+ * Compatibility for iOS <= 15.x
+ *
+ * @package automattic/jetpack-search
+ */
+namespace Automattic\Jetpack\Search\Compatibility;
+
+add_filter( 'jetpack_search_classic_search_enabled', __NAMESPACE__ . '\enable_classic_search_for_unsupported_browsers', 10, 1 );
+
+/**
+ * Enable Classic Search when the frontend is not working.
+ *
+ * @param boolean $clasic_search_enabled Whether Classic Search is enabled.
+ */
+function enable_classic_search_for_unsupported_browsers( $clasic_search_enabled ) {
+	if ( $clasic_search_enabled ) {
+		return true;
+	}
+	$user_agent = isset( $_SERVER['HTTP_USER_AGENT'] ) ? filter_var( wp_unslash( $_SERVER['HTTP_USER_AGENT'] ) ) : '';
+	// Generated by `browserslist-useragent-regexp`
+	// TODO: make the value re-generated on each build.
+	// !! The regex doesn't really work!!
+	return ! preg_match( '/Edge?\/1{2}[6-8](\.0|)(\.\d+|)|Firefox\/1{2}[78]\.0(\.\d+|)|Chrom(ium|e)\/(109\.0|1{2}[5-8]\.0)(\.\d+|)|Maci.+ Version\/(16\.6|17\.0)([,.]\d+|)( \(\w+\)|)( Mobile\/\w+|) Safari\/|Chrome.+OPR\/10[23]\.0\.\d+|(CPU[ +]OS|iPhone[ +]OS|CPU[ +]iPhone|CPU IPhone OS|CPU iPad OS)[ +]+(15[._][67]|16[._]6|17[._]0)([._]\d+|)|Opera Mini|Android:?[ /-]1{2}7(\.0|)(\.\d+|)|Android.+Chrom(ium|e)\/1{2}7\.0(\.\d+|)|SamsungBrowser\/2{2}\.0/i', $user_agent );
```

Comment:
@bluefuton  If the regex works well then this would be an ideal solution, but it doesn't 🤔

Reply:
I test with latest Safari and Chrome - they both failed the regex 😢